keydata.go

```diff
@@ -52,6 +55,92 @@ const (
 	AuthModePIN
 )
 
+type afSplitDataRawHdr struct {
+	Stripes uint32
+	HashAlg tpm2.HashAlgorithmId
+	Size    uint32
+}
+
+type afSplitDataRaw struct {
+	Hdr  afSplitDataRawHdr
+	Data tpm2.RawBytes
+}
+
+func (d *afSplitDataRaw) Marshal(w io.Writer) (nbytes int, err error) {
+	return tpm2.MarshalToWriter(w, d.Hdr, d.Data)
+}
+
+func (d *afSplitDataRaw) Unmarshal(r io.Reader) (nbytes int, err error) {
+	var h afSplitDataRawHdr
+	n, err := tpm2.UnmarshalFromReader(r, &h)
+	nbytes += n
+	if err != nil {
+		return nbytes, xerrors.Errorf("cannot unmarshal header: %w", err)
+	}
+
+	data := make([]byte, h.Size)
```

Comment:
Are there any reasonable limits that can be placed on this allocation? Allocating four gigabytes may be a surprising operation. Allocating zero bytes here may also lead to a divide-by-zero later on.

Reply:
There are issues like this in other places too. I've pushed a fix to go-tpm2 to ensure that sizes that are larger than the amount of data remaining are rejected, but that requires an update the vendored go-tpm2.

Zero bytes here will trigger a EOF in keyData.Unmarshal. The call to makeAfSplitData is called during marshalling instead.